checkov/terraform/checks/graph_checks/aws/IAMManagedIAMFullAccessPolicy.yaml

```diff
@@ -0,0 +1,42 @@
+---
+metadata:
+  id: "CKV2_AWS_51"
+  name: "Ensure AWS Managed IAMFullAccess IAM policy is not used."
+  category: "IAM"
+  severity: "high"
+definition:
+  not:
+    or:
+      - cond_type: attribute
+        resource_types:
+        - "data.aws_iam_policy"
+        attribute: "name"
+        operator: equals
+        value: "IAMFullAccess"
+      - cond_type: attribute
+        resource_types:
+        - "data.aws_iam_policy"
+        attribute: "arn"
+        operator: contains
+        value: "IAMFullAccess" 
+      - cond_type: attribute
+        resource_types:
+        - "aws_iam_policy_attachment"
+        - "aws_iam_user_policy_attachment"
+        - "aws_iam_role_policy_attachment"
+        - "aws_iam_group_policy_attachment"
+        attribute: "policy_arn"
+        operator: contains
+        value: "IAMFullAccess"
+      - cond_type: attribute
```

Comment:
the test failed in a different array, because `managed_policy_arns` in `aws_iam_role` is not required, therefore the check should also pass, when it doesn't exists.
ex.
```hcl
resource "aws_iam_role" "example" {
  name = "example"

  assume_role_policy = jsonencode({
    Version = "2012-10-17"

    Statement = [{
      Effect = "Allow"
      Action = "sts:AssumeRole"

      Principal = {
        Service = "codebuild.amazonaws.com"
      }
    }]
  })
}
```

Reply:
added 2 more tests for `aws_iam_role`.  One that passes without `managed_policy_arns` and another that fails with multiple items in the array, where it contains the value in the check `IAMFullAccess'. 

@gruebel let me know if that is what you were looking for, as it was not clear what you meant by "the test failed in a different array", noting that I did not have to change my check for the additional tests to work as expected.

Thanks!

Reply:
I don't know why I wrote 'array' I meant area 😄

Reply:
yeah, you are right about the change. let's see what the unit tests tell us, otherwise I need to recheck, why a test in different part failed.